### ISSUE TITLE
SDKRuntimeException: Challenge presentation failure

### ISSUE THREAD

**Issue by mikkoville** [Adyen/adyen-3ds2-android#37]
**Describe the bug**
We are seeing in our crash reporting tool users getting fatal exception of:

```
Caused by com.adyen.threeds2.exception.SDKRuntimeException: Challenge presentation failure.
       at atd.y.c.a(c.java:2)
       at atd.y.c.a(c.java:1)
       at com.adyen.threeds2.internal.ui.activity.ChallengeActivity.a(ChallengeActivity.java:40)
       at com.adyen.threeds2.internal.ui.activity.ChallengeActivity.onCreate(ChallengeActivity.java:17)
       at android.app.Activity.performCreate(Activity.java:8198)
       at android.app.Activity.performCreate(Activity.java:8182)
       at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1309)
       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3765)
       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3968)
       at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85)
       at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
       at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2307)
       at android.os.Handler.dispatchMessage(Handler.java:106)
       at android.os.Looper.loop(Looper.java:246)
       at android.app.ActivityThread.main(ActivityThread.java:8512)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:602)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1130)
```

```
Fatal Exception: java.lang.RuntimeException: Unable to start activity ComponentInfo{com.xxx.xxx/com.adyen.threeds2.internal.ui.activity.ChallengeActivity}: com.adyen.threeds2.exception.SDKRuntimeException: Challenge presentation failure.
       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3792)
       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3968)
       at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85)
       at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
       at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2307)
       at android.os.Handler.dispatchMessage(Handler.java:106)
       at android.os.Looper.loop(Looper.java:246)
       at android.app.ActivityThread.main(ActivityThread.java:8512)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:602)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1130)
```

**To Reproduce**
- Not been able to reproduce locally

**Expected behavior**
n/a

**Screenshots**
n/a

**Desktop (please complete the following information):**
 n/a

**Smartphone (please complete the following information):**
 - Device: [Galaxy S10+, Galaxy A70, Mi 10T Lite, Galaxy A21s, Galaxy A02, Galaxy A50, Mi 10T Pro, ]
 - OS: [Android 10 (14%), Android 11 (86%)]
 - Version [com.adyen.threeds:adyen-3ds2:2.2.2]

**Additional context**
Do you have any ideas what could be causing this?

**Comment by caiofaustino** [comment]
Hey @mikkoville. Yeah that error is sent when an unexpected Intent started the Activity so it fails for added security. But I agree it shouldn't cause a crash in the app. I'll add a fix for this.

**Comment by mikkoville** [comment]
@caiofaustino Thank you! So is this order of calls issue on our end? That in some edge case we try to do the challenge in some invalid state?

**Comment by caiofaustino** [comment]
Hmmm, that's hard to say. The SDK is not supposed to handle process kills, so it could be the case that you have an instance of a Transaction that is not valid anymore or something like that, but I would need to know more about the scenario.

**Comment by mikkoville** [comment]
Right yes that is indeed possible even tough probably quite an edge case. Thanks

**Comment by caiofaustino** [comment]
The fix to this crash is in version `2.2.3`, now the Activity will simply close with an error.
